### PR TITLE
[embassy-nrf] IEEE 802.15.4 radio

### PR DESCRIPTION
**Pull request by blueluna**
Following the nRF BLE radio work done by @guissalustiano, I propose a IEEE 802.15.4 radio.

This work is a concoction of IEEE radio work from the nrf-hal, old code of mine, ...

This work has been tested on micro:bit v2 (nRF52833) and Makerdiary nRF52840-MDK. Following examples use this work,
 - https://github.com/blueluna/embassy-nrf-ieee802154
 - https://github.com/blueluna/ieee802154-sniffer

BLE was tested with a slightly modified version of Jewel.

### CODE REVIEW

**Review by lulf — APPROVED** [review]
Thanks for the addition, it looks reasonable to me. Did you test on nRF53 as well?

**Comment by Dirbaio** [comment]
could you add `impl_radio!` in `chips/` to the other chips, not just nrf52833?

**Comment by blueluna** [comment]
> Thanks for the addition, it looks reasonable to me. Did you test on nRF53 as well?

Thanks. I have not tested on nRF53. I don have access to a nRF53.

> Could you add impl_radio! in chips/ to the other chips, not just nrf52833?

Sure. nRF52840 has already been added in the BLE PR.

**Comment by guissalustiano** [comment]
> could you add impl_radio! in chips/ to the other chips, not just nrf52833?
This requires some changes on the BLE part, I opened a PR (#2662)[https://github.com/embassy-rs/embassy/pull/2662) with the changes.
You can also merge here and close the other PR

**Comment by blueluna** [comment]
I did something similar just more cfg feature blocks.

One thing that keeping me from adding nRF51 to BLE is following,
https://github.com/embassy-rs/embassy/blob/60567b72fb8807bf72aa28f0e2c551953c1ef7bf/embassy-nrf/src/util.rs#L37
I do not understand why it is not implemented for nRF51.

File: embassy-nrf/src/radio/mod.rs [hunk 1]
```diff
@@ -6,12 +6,42 @@
 #![macro_use]
 
 /// Bluetooth Low Energy Radio driver.
+#[cfg(not(feature = "nrf51"))]
 pub mod ble;
+#[cfg(any(
+    feature = "nrf52820",
+    feature = "nrf52833",
+    feature = "nrf52840",
+    feature = "_nrf5340-net"
```
Comment (guissalustiano) [inline embassy-nrf/src/radio/mod.rs:15]: The nrf52811 also supports the IEE 802.15.4 according to [the datasheet](https://infocenter.nordicsemi.com/pdf/nRF52811_PS_v1.0.pdf) (6.14.13)

**Comment by blueluna** [comment]
I re-ran `embassy-nrf-ieee802154` on nrf52833 and nrf52840. Jewel beacon on **nrf51822**, nrf52833 and nrf52840. So `slice_in_ram_or` seems to work on nrf51.

**Review by guissalustiano — APPROVED** [review]

**Pull request merged**